lib/transaction.d.ts

```diff
@@ -1,25 +1,50 @@
 import { Logging } from './model'
 import { Promise } from './promise'
-import { Sequelize } from './sequelize'
-
-/**
- * The transaction object is used to identify a running transaction. It is created by calling
- * `Sequelize.transaction()`.
- *
- * To run a query under a transaction, you should pass the transaction in the options object.
- */
-export class Transaction {
-    constructor(sequelize: Sequelize, options: TransactionOptions)
+import { Deferrable, Sequelize } from './sequelize'
 
+// tslint:disable-next-line
+export namespace Transaction {
     /**
-     * Commit the transaction
+     * Isolations levels can be set per-transaction by passing `options.isolationLevel` to `sequelize.transaction`.
+     * Default to `REPEATABLE_READ` but you can override the default isolation level by passing `options.isolationLevel` in `new Sequelize`.
+     *
+     * The possible isolations levels to use when starting a transaction:
+     *
+     * ```js
+     * {
+     *   READ_UNCOMMITTED: "READ UNCOMMITTED",
+     *   READ_COMMITTED: "READ COMMITTED",
+     *   REPEATABLE_READ: "REPEATABLE READ",
+     *   SERIALIZABLE: "SERIALIZABLE"
+     * }
+     * ```
+     *
+     * Pass in the desired level as the first argument:
+     *
+     * ```js
+     * return sequelize.transaction({isolationLevel: Sequelize.Transaction.SERIALIZABLE}, transaction => {
+     *
+     *  // your transactions
+     *
+     * }).then(result => {
+     *   // transaction has been committed. Do something after the commit if required.
+     * }).catch(err => {
+     *   // do something with the err.
+     * });
+     * ```
      */
-    public commit(): Promise<void>
+    const enum ISOLATION_LEVELS {
```

Comment:
imo these should just be exported top-level, which is what we do in other files if Sequelize provides both.

Reply:
What do you mean with `top-level`, can you provide an example?